File: 1636. Sort Array by Increasing Frequency/frequencySort.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const frequencySort = (nums) => {
  const frequencyOfNums = {};
  for (let i = 0; i < nums.length; i++) {
    if (!frequencyOfNums[nums[i]]) {
      frequencyOfNums[nums[i]] = 0;
    }
    frequencyOfNums[nums[i]]++;
  }

  const sortedNums = nums.sort((a, b) => {
    return (frequencyOfNums[a] - frequencyOfNums[b]) || b - a
  });
  
  return sortedNums;
};


const nums = [1, 1, 2, 2, 2, 3];
console.log('frequencySort: ', frequencySort(nums));

File: 1636. Sort Array by Increasing Frequency/frequencySort.py
from typing import List

class Solution:
  def frequencySort(self, nums: List[int]) -> List[int]:
    frequencyOfNums = {}
    for n in nums:
      if n not in frequencyOfNums:
        frequencyOfNums[n] = 0
      frequencyOfNums[n] += 1
      
    sortedNums = sorted(nums, key = lambda x: (frequencyOfNums[x], -x))
    
    return sortedNums
    
    
nums = [1, 1, 2, 2, 2, 3]
solution = Solution()
print('frequencySort: ', solution.frequencySort(nums))
